File: VECTOAux/VectoAuxiliaries/Electrics/IAlternator.vb

Namespace Electrics

    'Used By Combined Alternator.
    'Model based on CombinedALTS_V02_Editable.xlsx
    Public Interface IAlternator

        'D6
        Property AlternatorName As String

        'G6
        Property PulleyRatio As Single

        'S9
        ReadOnly Property SpindleSpeed As Double

        'S10
        ReadOnly Property Efficiency As Double

        'C10-D15
        Property InputTable2000 As List(Of AltUserInput)

        'F10-G15
        Property InputTable4000 As List(Of AltUserInput)

        'I10-J15
        Property InputTable6000 As List(Of AltUserInput)

        'M10-N15
        Property RangeTable As List(Of Table4Row)

        'Test Equality
        Function IsEqualTo(other As IAlternator) As Boolean


    End Interface


End Namespace

File: VECTOAux/BusAuxiliaries/Interfaces/DownstreamModules/Electrics/IAlternator.cs
using System.Collections.Generic;

namespace DownstreamModules.Electrics
{
	public class Table4Row
	{
		public double RPM;
		public double Efficiency;

		public Table4Row(double rpm, double eff)
		{
			this.RPM = rpm;
			this.Efficiency = eff;
		}
	}

	// Used By Combined Alternator.
	// Model based on CombinedALTS_V02_Editable.xlsx
	public interface IAlternator
	{
		// D6
		string AlternatorName { get; set; }

		// G6
		double PulleyRatio { get; set; }

		// S9
		double SpindleSpeed { get; }

		// S10
		double Efficiency { get; }

		// C10-D15
		List<AltUserInput> InputTable2000 { get; set; }

		// F10-G15
		List<AltUserInput> InputTable4000 { get; set; }

		// I10-J15
		List<AltUserInput> InputTable6000 { get; set; }

		// M10-N15
		List<Table4Row> RangeTable { get; set; }

		// Test Equality
		bool IsEqualTo(IAlternator other);
	}
}
